core/server/worker/src/main/java/alluxio/worker/grpc/GrpcExecutors.java

```diff
@@ -48,6 +48,12 @@
           THREAD_STOP_MS, TimeUnit.MILLISECONDS, new SynchronousQueue<>(),
           ThreadFactoryUtils.build("BlockDataReaderExecutor-%d", true)));
 
+  public static final ExecutorService BLOCK_READER_SERIALIZED_RUNNER_EXECUTOR =
+      new ImpersonateThreadPoolExecutor(new ThreadPoolExecutor(THREADS_MIN,
+          ServerConfiguration.getInt(PropertyKey.WORKER_NETWORK_BLOCK_READER_THREADS_MAX),
```

Comment:
What does `WORKER_NETWORK_BLOCK_READER_THREADS_MAX` limit when used with this pool?

Reply:
I used that value because it's the same one that corresponds to the `BlockDataReaderExecutor`. We should never have more task executors than we have threads from the `BlockDataReaderExecutor`

Reply:
We will have more streams than we have reader-threads.
Not all streams are in a state to a read block, so total count should be allowable to be more than that. Otherwise, we're stalling completions/cancellations until readers are finished.

Reply:
updated